File: app/src/main/java/com/semisonfire/cloudgallery/ui/photo/PhotoDetailContract.java
package com.semisonfire.cloudgallery.ui.photo;

import android.net.Uri;

import com.semisonfire.cloudgallery.data.model.Photo;
import com.semisonfire.cloudgallery.ui.base.MvpPresenter;
import com.semisonfire.cloudgallery.ui.base.MvpView;

public interface PhotoDetailContract {

    interface View extends MvpView {

        void onPhotoDownloaded(String path);

        void onFilePrepared(Uri uri);

        void onFilesChanged(Photo photo);
    }

    interface Presenter<V extends View> extends MvpPresenter<V> {

        void download(Photo photo);

        void delete(Photo photo, int from);

        void restore(Photo photo);

    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/photo/PhotoDetailContract.kt
package com.semisonfire.cloudgallery.ui.photo

import android.graphics.Bitmap
import android.net.Uri

import com.semisonfire.cloudgallery.data.model.Photo
import com.semisonfire.cloudgallery.core.mvp.MvpPresenter
import com.semisonfire.cloudgallery.core.mvp.MvpView

interface PhotoDetailContract {

  interface View : MvpView {

    fun onPhotoDownloaded(path: String)

    fun onFilePrepared(uri: Uri)

    fun onFilesChanged(photo: Photo)
  }

  interface Presenter : MvpPresenter<View> {

    fun download(photo: Photo)

    fun delete(photo: Photo, from: Int)

    fun restore(photo: Photo)

    fun createShareFile(bitmap: Bitmap)
  }
}
